stix/typedb_inserter.py

```diff
@@ -0,0 +1,71 @@
+#
+# Copyright (C) 2022 Vaticle
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+import logging
+from functools import partial
+from multiprocessing.dummy import Pool as ThreadPool
+from timeit import default_timer as timer
+
+from typedb.client import *
+
+
+def data_count(uri, database):
+    with TypeDB.core_client(uri) as client:
+        with client.session(database, SessionType.DATA) as session:
+            with session.transaction(TransactionType.READ) as tx:
+                return tx.query().match_aggregate("match $x isa thing; count;").get().as_int()
+
+
+class TypeDBInserter:
```

Comment:
created a class for the batching inserter, which is used as a class from outside (this way the client can be reused internally very easily without passing the uri/database/batch/threads everywhere)